packages/react-components/react-table/src/components/DataGrid/DataGrid.test.tsx

```diff
@@ -34,6 +39,16 @@ describe('DataGrid', () => {
       items: testItems,
       columns: testColumns,
     },
+    testOptions: {
+      'consistent-callback-args': {
+        ignoreProps: ['onColumnResize'],
```

Comment:
onColumnResize is not a real, browser event

Reply:
IMO `setColumnWidth` should support an event, it should only be called on user action. Especially since in all our examples `setColumnWidth` is actually called by a mouse event. We should let users have the most relevant DOM event so that they have more control over interaction flow.

It also aligns with other handlers like `onSelectionChange` or `onSortChange`

Reply:
Actually `setColumnWidth` is used in the `ResizableColumnsUncontrolled` story. Calling this programmatically (and not from a mouseMove) causes the `onColumnResize` callback to trigger too, and there is no event to pass.

We could potentially pass `undefined` in that case. This would be a breaking change to the existing api though, but since this is unstable/preview, no big deal probably, just something to keep in mind.

Reply:
I'm OK with an explicit undefined in this case